fsharp-backend/src/HttpMiddleware/HttpMiddlewareV1.fs

```diff
@@ -0,0 +1,64 @@
+/// Thin, bytes-friendly middleware
+///
+/// This Middleware corresponds with `HTTPBytes` handlers
+[<RequireQualifiedAccess>]
+module HttpMiddleware.HttpMiddlewareV1
+
+module Request =
+  open Prelude
+  open LibExecution.VendoredTablecloth
+
+  module RT = LibExecution.RuntimeTypes
+  module HttpQueryEncoding = BackendOnlyStdLib.HttpQueryEncoding
+
+  let private parseHeaders (headers : (string * string) list) =
+    headers
+    |> List.map (fun (k, v) -> (String.toLowercase k, RT.DStr v))
+    |> Map
+    |> RT.Dval.DObj
+
+  let fromRequest
+    (uri : string)
+    (headers : List<string * string>)
+    (query : string)
+    (body : byte array)
+    : RT.Dval =
+    [ "body", RT.DBytes body
+      "queryParams", RT.DStr query
+      "headers", parseHeaders headers
+      "url", RT.DStr uri ]
+    |> RT.Dval.obj
+
+
+module Response =
+  open Prelude
+
+  module RT = LibExecution.RuntimeTypes
+
+  type HttpResponse =
+    { statusCode : int
+      body : byte array
+      headers : HttpHeaders.T }
+
+  module DvalReprLegacyExternal = LibExecution.DvalReprLegacyExternal
+  module Telemetry = LibService.Telemetry
+
+  let toHttpResponse (result : RT.Dval) : HttpResponse =
+    match result with
+    // Expected user responses
+    | RT.DHttpResponse (RT.Redirect str) ->
+      Telemetry.addTags [ "response-type", "httpResponse redirect" ]
+      { statusCode = int 302; headers = [ "Location", str ]; body = [||] }
+
+    | RT.DHttpResponse (RT.Response (code, headers, RT.DBytes body)) ->
+      Telemetry.addTags [ "response-type", "httpResponse response" ]
+      { statusCode = int code; headers = headers; body = body }
```

Comment:
TODO: consider tolowercase-ing the response headers here

Reply:
I had thought to remove RT.redirect here as well. But I guess since it has no body, it's fine. This is making me think that RT.redirect shouldn't exist, and should just be encoded as an RT.response.

(nothing for this PR, just thinking out loud)

Reply:
> making me think that RT.redirect shouldn't exist, and should just be encoded as an RT.response.

Agreed.